gadget/install/partition.go

```diff
@@ -31,16 +33,32 @@ import (
 	"github.com/snapcore/snapd/gadget/quantity"
 	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/osutil/disks"
 )
 
 var (
 	ensureNodesExist = ensureNodesExistImpl
 )
 
+// reloadPartitionTable reloads the partition table depending on what the gadget
+// says - if the gadget has a special marker file, then we will use a special
+// reload mechanism, implemented for a specific device.
+func reloadPartitionTable(gadgetRoot string, device string) error {
+	if osutil.FileExists(filepath.Join(gadgetRoot, "force-partition-table-reload-via-device-rescan.txt")) {
```

Comment:
I wouldn't use the innocent-looking ".txt" extension for this; maybe ".flag"?

But looking at the rest of the code, it seems to me it would be much simpler if we cheked for the presence of a `partition-table-reload` executable file instead. Then we would just run it passing the `device` as parameter.

It would make things simpler for us to test, and it seems like a smaller hack to me.

Reply:
Or even we could unify the code paths, if insread of hardcoding `partx` we first tried to read the command name from the environment.

Reply:
yeah the name of this file is totally arbitrary, we have done this before, and I believe we used .txt file extension before.

I'd rather not have us execute stuff from the gadget though, while it seems like less of a support burden for us, it's introducing more of an interface between snapd and the gadget which we don't want since this really is meant to live for one specific device and there's more of a chance that other gadgets/project use this and makes it harder for us to delete this code later on (if we ever can do so).

Reply:
I suggested something like with some reluctance but is the better compromise. Anyway if we were running something the gadget it would have to be a real hook, and it seems overkill, problematic to carry forwared as Ian mentioned.

In terms of look of the marker file the closest we have is:  https://github.com/snapcore/snapd/blob/master/bootloader/grub.go#L196

so "force-*" is good, not externsion, and under meta (no super strong opinion on the latter but doesn't seem bad either)

Reply:
done, I dropped the .txt file extension and moved this under meta